src/main/java/org/broadinstitute/consent/http/service/LibraryCardService.java

```diff
@@ -66,9 +73,14 @@ public LibraryCard updateLibraryCard(LibraryCard libraryCard, Integer id, Intege
         return this.libraryCardDAO.findLibraryCardById(id);
     }
 
-    public void deleteLibraryCardById(Integer id) {
-        LibraryCard libraryCard = this.libraryCardDAO.findLibraryCardById(id);
-        throwIfNull(libraryCard);
+    public void deleteLibraryCardById(Integer id, User user) {
+        Boolean isAdmin = checkIsAdmin(user);
+        LibraryCard card = findLibraryCardById(id);
+        throwIfNull(card);
+        //If user is not an admin and LC institutionID doesn't match the users's throw a ForbiddenException
+        if(!isAdmin && !card.getInstitutionId().equals(user.getInstitutionId())) {
+            throw new ForbiddenException("You are not authorized to delete this library card");
+        }
```

Comment:
We should move the authorization logic in the resource layer which is currently where we enforce these kinds of restrictions. At this layer, we're just concerned with business logic.